### PR TITLE
Add test to check older pickles continue to unpickle in future

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This adds a test that we can un-pickle pickles made with the current version of IO in the future. I feel we do claim this should work, but there was no test for it.

We could hypothetically add more tests like this, but this is at least a start, and a check for future.

### CODE REVIEW

File: tst/pickle.g [hunk 1]
```diff
@@ -244,3 +244,58 @@ elif ConjugacyClasses( UnderlyingGroup( tbl ) )
   Error( 42 );
 fi;
 
+# Check we can unpickle old pickled data
+# Geneated with GAP 4.12.0, IO 4.7.2 on 2022-09-08
```
Comment (fingolfin) [inline tst/pickle.g:248]:
```suggestion
# Generated with GAP 4.12.0, IO 4.7.2 on 2022-09-08
```

File: tst/pickle.g [hunk 2]
```diff
@@ -244,3 +244,58 @@ elif ConjugacyClasses( UnderlyingGroup( tbl ) )
...
+\>8Z(257)^0FFEL\<12ZmodnZObj(3,65537)FFEL\<12ZmodnZObj(1,65537)FFEL\<12ZmodnZObj(0,65537)FFEL\>AZ(65537,2)FFEL\<12Zmod\
+nZObj(1,65537)FFEL\<12ZmodnZObj(0,65537)PERM\>9(1,2,3,4)GAPL\>4UPOLINTG\>11ILIS\>2ILIS\>3INTG\>11INTG\>11INTG\>11INTG\
+\>10INTG\>11UPOLINTG\>11ILIS\>2ILIS\>8INTG\>11INTG\>10INTG\>10INTG\>11INTG\>10INTG\>10INTG\>10INTG\>11INTG\>2-3INTG\>1\
+1URFUINTG\>11ILIS\>3ILIS\>2INTG\>11INTG\>11ILIS\>2INTG\>12INTG\>11INTG\>10INTG\>11MREC\>2ISTR\>1bMSTR\>3MaxISTR\>1aINT\
+G\>11IREC\>2ISTR\>1cINTG\>13ISTR\>1dISTR\>4TillMF8V\>3FFEL\>4Z(5)FFEL\>60*Z(5)FFEL\>6Z(5)^2IF8V\>3FFEL\>4Z(5)FFEL\>60*\
+Z(5)FFEL\>6Z(5)^2MF8V\>3FFEL\>4Z(7)FFEL\>60*Z(7)FFEL\>6Z(7)^2IF8V\>3FFEL\>4Z(7)FFEL\>60*Z(7)FFEL\>6Z(7)^2MF2V\>2FFEL\>\
+6Z(2)^0FFEL\>60*Z(2)IF2V\>2FFEL\>6Z(2)^0FFEL\>60*Z(2)MF8M\>1MF8V\>3FFEL\>4Z(5)FFEL\>60*Z(5)FFEL\>6Z(5)^2IF8M\>1MF8V\>3\
+FFEL\>4Z(5)FFEL\>60*Z(5)FFEL\>6Z(5)^2MF8M\>1MF8V\>3FFEL\>4Z(7)FFEL\>60*Z(7)FFEL\>6Z(7)^2IF8M\>1MF8V\>3FFEL\>4Z(7)FFEL\
+\>60*Z(7)FFEL\>6Z(7)^2MF2M\>1MF2V\>2FFEL\>6Z(2)^0FFEL\>60*Z(2)IF2M\>1MF2V\>2FFEL\>6Z(2)^0FFEL\>60*Z(2)SREF\>1MREC\>3IS\
+TR\>1xINTG\>11ISTR\>1lSREF\>1ISTR\>1rSREF\<33";
```
Comment (fingolfin) [inline tst/pickle.g:275]: Any particular reason to store this as a string instead of storing it as a pickle file?

Comment (ChrisJefferson) [reply tst/pickle.g:275]: No, it saved making an extra file, but that would probably be easier.

**Comment by fingolfin** [comment]
The macos test fails (see issue #105), but this is not caused by this PR so I'll merge anyway.

**Pull request merged**